# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/Http2ConnectTransaction.cpp
# Commit: ee9744982673
# Full Hash: ee974498267381418f06efd6ab92c1f2c5f3735d
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2022-11-15 16:44:51
# Regressor Bug: 1791633
# File Overlap Count: 1
# Description:
#   Bug 1791633 - separate nsITLSSocketControl from nsITransportSecurityInfo r=necko-reviewers,kershaw,jschanck
#   
#   Depends on D160311
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160313
# ==============================================================================

diff -r f5a4bfdaba40 -r ee9744982673 netwerk/protocol/http/Http2ConnectTransaction.cpp
--- a/netwerk/protocol/http/Http2ConnectTransaction.cpp	Tue Nov 15 05:34:40 2022 +0000
+++ b/netwerk/protocol/http/Http2ConnectTransaction.cpp	Tue Nov 15 05:34:40 2022 +0000
@@ -30,7 +30,7 @@
 #include "prio.h"
 
 #if defined(FUZZING)
-#  include "FuzzySecurityInfo.h"
+#  include "FuzzySocketControl.h"
 #  include "mozilla/StaticPrefs_fuzzing.h"
 #endif
 
@@ -666,7 +666,7 @@
     // target is on the socket thread. That's all we really care about.
     nsCOMPtr<nsIEventTarget> sts =
         do_GetService(NS_SOCKETTRANSPORTSERVICE_CONTRACTID);
-    MOZ_ASSERT((!target && !callback) || (target == sts));
+    MOZ_ASSERT(!target || (target == sts));
     if (target && (target != sts)) {
       return NS_ERROR_FAILURE;
     }
@@ -891,7 +891,7 @@
     // target is on the socket thread. That's all we really care about.
     nsCOMPtr<nsIEventTarget> sts =
         do_GetService(NS_SOCKETTRANSPORTSERVICE_CONTRACTID);
-    MOZ_ASSERT((!target && !callback) || (target == sts));
+    MOZ_ASSERT(!target || (target == sts));
     if (target && (target != sts)) {
       return NS_ERROR_FAILURE;
     }